### ISSUE TITLE
Unhandled exception when deploying from a private GitHub repo via SSH

### ISSUE THREAD

**Issue by AlexIoannides** [bodywork-ml/bodywork-core#108]
When trying to deploy a private repo via SSH using,
```
$ bodywork deployment create \
    --namespace=arc-cpre \
    --name=d1 \
    --git-repo-url=git@github.com/everlution/arc-cpre.git \
    --git-repo-branch=rest-api-definition \
    -L
```

I got the following unhandled exception:

```
testing with local workflow-controller - retries are inactive
namespace=arc-cpre is setup for use by Bodywork
2021-06-22 19:48:29,882 - INFO - workflow_execution.run_workflow - attempting to run workflow for project=git@github.com/everlution/arc-cpre.git on branch=rest-api-definition in kubernetes namespace=arc-cpre
2021-06-22 19:48:29,939 - ERROR - workflow_execution.run_workflow - failed to execute workflow for rest-api-definition branch of project repository at git@github.com/everlution/arc-cpre.git: Unable to setup SSH for Github and you are trying to connect via SSH: 'failed to setup SSH for github.com - cannot find BODYWORK_GIT_SSH_PRIVATE_KEY environment variable'
Traceback (most recent call last):
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/git.py", line 63, in download_project_code_from_repo
    setup_ssh_for_git_host(hostname)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/git.py", line 137, in setup_ssh_for_git_host
    raise KeyError(msg)
KeyError: 'failed to setup SSH for github.com - cannot find BODYWORK_GIT_SSH_PRIVATE_KEY environment variable'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/workflow_execution.py", line 81, in run_workflow
    download_project_code_from_repo(repo_url, repo_branch, cloned_repo_dir)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/git.py", line 74, in download_project_code_from_repo
    raise BodyworkGitError(msg)
bodywork.exceptions.BodyworkGitError: Unable to setup SSH for Github and you are trying to connect via SSH: 'failed to setup SSH for github.com - cannot find BODYWORK_GIT_SSH_PRIVATE_KEY environment variable'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/workflow_execution.py", line 156, in run_workflow
    if config.project.run_on_failure and type(e) not in [
UnboundLocalError: local variable 'config' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/workflow_execution.py", line 167, in run_workflow
    f"Error executing failure stage: {config.project.run_on_failure}"
UnboundLocalError: local variable 'config' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/bin/bodywork", line 8, in <module>
    sys.exit(cli())
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/cli/cli.py", line 302, in cli
    args.func(args)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/cli/cli.py", line 324, in wrapper
    func(*args, **kwargs)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/cli/cli.py", line 393, in deployment
    workflow(pass_through_args)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/cli/cli.py", line 324, in wrapper
    func(*args, **kwargs)
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/cli/cli.py", line 562, in workflow
    run_workflow(
  File "/Users/alexioannides/Dropbox/bodywork_client_repos/arc-CPRE/.venv/lib/python3.8/site-packages/bodywork/workflow_execution.py", line 176, in run_workflow
    if config is not None and config.project.usage_stats:
UnboundLocalError: local variable 'config' referenced before assignment
```

**Comment by AlexIoannides** [comment]
This happened when I was running the workflow-controller locally with a private repo, but had forgot to set `BODYWORK_GIT_SSH_PRIVATE_KEY`. This caused an exception to be thrown [here](https://github.com/bodywork-ml/bodywork-core/blob/1ce9323c0a365158e308cb0c6accb0df90fe51ef/src/bodywork/workflow_execution.py#L81), which is prior to when `config` is set. Which the caused a cascade of exceptions...

**Comment by AlexIoannides** [comment]
Initial thoughts:
- we need to handle `bodywork.exceptions.BodyworkGitError` separately from the others.
- I'm starting to come around to your way of thinking, which is to always look in the local `.ssh/id_rsa` for the key and then find a way to create this in the remote containers. Although effective, it's just too flakey at the moment.

**Comment by Marios85** [comment]
Simple solution for this, check the exception first and then the config in the catch block. I will make the change now.

As for using the local .ssh, it will make it easier for people. If I have my Git provider configured, then in theory it means I will have it already configured for Bodywork to access the repo if you use that directory.

**Comment by Marios85** [comment]
Fixed in V2.1.5
